### PR TITLE
adding new providerName config

### LINKED ISSUE #376: Add a config property to append a string to the user agent.
**## Problem Statement**
To determine whether connections and records are coming from Confluent Cloud, we will need to append a string to the user agent.

**## Proposed Solution**
It could be a config property called "provider" with null by default.

## Next Steps

- [ ] Team consensus to proceed
- [ ] Schedule Design Session
- [ ] Complete Design Review

### PR DESCRIPTION
**Pull request by ryancrawcour**
## Type of PR

- [ ] Documentation changes
- [x] Code changes
- [x] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
adding new provider config to allow us to differentiate between ConfluentCloud managed, and self managed. 

## Issues Closed or Referenced
- Closes #376

### CODE REVIEW

**Comment by ryancrawcour** [comment]
@NathanNam this addresses #376. could you please take a look and let me know if this is what you were after. 
i have added the config item. default to null. it isn't on the UX (because we don't really want customers to be setting it)
i have not yet added it to our user-agent string, @moderakh / @kushagraThapar should i?

**Review by moderakh — APPROVED** [review]
Thanks @ryancrawcour LGTM.

File: src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java [hunk 1]
```diff
@@ -32,12 +33,16 @@
         + "For example: topic1#con1,topic2#con2.";
     private static final String COSMOS_CONTAINER_TOPIC_MAP_DISPLAY = "Topic-Container map";
 
+    public static final String  COSMOS_PROVIDER_NAME_CONF = "connect.cosmos.provider.name";
```
Comment (moderakh) [inline src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:36]: how would differentiating between different apps work? do we need to differentiate the apps?

Comment (moderakh) [reply src/main/java/com/azure/cosmos/kafka/connect/CosmosDBConfig.java:36]: discussed offline. in future if we see the need, we can add another config later for app name (outside of the scope of this PR).

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**